contracts/sol6/katana/KatanaFeeHandler.sol

```diff
@@ -0,0 +1,459 @@
+ *     Burning KNC
+ *     Reward an address that staked knc in kyberStaking contract. AKA - stakers
+ *     Rebate reserves for supporting trades.
+ * @dev Code flow:
+ *    Per trade on kyberNetwork, it calls handleFees() function which
+ *    internally accounts for network & platform fees from the trade. 
+ *    Fee distribution:
+ *      rewards: send to fee pool
+ *      rebates: accumulated per rebate wallet, can be claimed any time.
+ *      burn: accumulated in the contract. Burned value and interval limited with safe check using
+ *        sanity rate.
+ *      Platfrom fee: accumulated per platform wallet, can be claimed any time.
+ */
+contract KatanaFeeHandler is Utils5, DaoOperator, ReentrancyGuard, IKyberFeeHandler {
+    using SafeMath for uint256;
+
+    uint256 internal constant DEFAULT_REWARD_BPS = 3000;
+    uint256 internal constant DEFAULT_REBATE_BPS = 3000;
+    uint256 internal constant SANITY_RATE_DIFF_BPS = 1000; // 10%
+
+    struct BRRData {
+        uint16 rewardBps;
+        uint16 rebateBps;
+    }
+
+    struct BRRWei {
+        uint256 rewardWei;
+        uint256 fullRebateWei;
+        uint256 paidRebateWei;
+        uint256 burnWei;
+    }
+
+    IKyberProxy public kyberProxy;
+    address public kyberNetwork;
+    address public feePool;
+    IERC20 public immutable knc;
+
+    uint256 public immutable burnBlockInterval;
+    uint256 public lastBurnBlock;
+
+    BRRData public brrData;
+
+    /// @dev amount of eth to burn for each burn knc call
+    uint256 public weiToBurn = 2 ether;
+
+    mapping(address => uint256) public feePerPlatformWallet;
+    mapping(address => uint256) public rebatePerWallet;
+    // total balance in the contract that is for rebate, reward, platform fee
+    uint256 public totalPayoutBalance;
+    /// @dev use to get rate of KNC/ETH to check if rate to burn knc is normal
+    /// @dev index 0 is currently used contract address, indexes > 0 are older versions
+    ISanityRate[] internal sanityRateContract;
+
+    event FeeDistributed(
+        IERC20 indexed token,
+        address indexed platformWallet,
+        uint256 platformFeeWei,
+        uint256 rewardWei,
+        uint256 rebateWei,
+        address[] rebateWallets,
+        uint256[] rebatePercentBpsPerWallet,
+        uint256 burnAmtWei
+    );
+
+    event BRRUpdated(uint256 rewardBps, uint256 rebateBps, uint256 burnBps);
+
+    event FeePoolUpdated(address feePool);
+
+    event RebatePaid(address indexed rebateWallet, IERC20 indexed token, uint256 amount);
+
+    event PlatformFeePaid(address indexed platformWallet, IERC20 indexed token, uint256 amount);
+
+    event KncBurned(uint256 kncTWei, IERC20 indexed token, uint256 amount);
+
+    event EthReceived(uint256 amount);
+
+    event BurnConfigSet(ISanityRate sanityRate, uint256 weiToBurn);
+
+    event KyberNetworkUpdated(address kyberNetwork);
+
+    event KyberProxyUpdated(IKyberProxy kyberProxy);
+
+    constructor(
```

Comment:
Please pass reward + rebate into the constructor since we don't use the default values after deployed

Reply:
Ok